### ISSUE TITLE
add loop vectorization infrastructure

### ISSUE THREAD

**Issue by gergo-** [oracle/fastr#208]
**TL;DR**

Make automatic loop vectorization available for just-in-time (JIT) compilations in GraalVM Community Edition.

**Goals**

Add source compiler phases and supporting infrastructure to enable automatic loop vectorization in GraalVM CE JIT compilations. Users will be able to write natural Java loops and obtain efficient SIMD code for supported loop shapes. This will include many loops that are not vectorized by HotSpot's C2 compiler.

See the blog post on [Enhanced Automated Vectorization](https://medium.com/graalvm/enhanced-automated-vectorization-in-graalvm-76cd99925b6d) for some examples of automatic vectorization in GraalVM.

**Non-Goals**
